Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.cpp

```diff
@@ -343,13 +347,106 @@ namespace AzToolsFramework
                 .SetTip(PaintbrushDecreaseHardnessDesc)
                 .SetEntityComponentIdPair(m_ownerEntityComponentId)
                 .SetCallback(
-                    [this]()
+                    []()
                     {
                         AdjustHardnessPercent(-ed_paintBrushHardnessPercentAdjustAmount);
                     }),
         };
     }
 
+    void PaintBrushManipulator::RegisterActions()
+    {
+        auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(actionManagerInterface, "PaintBrushManipulator - could not get ActionManagerInterface on RegisterActions.");
+
+        auto hotKeyManagerInterface = AZ::Interface<AzToolsFramework::HotKeyManagerInterface>::Get();
+        AZ_Assert(hotKeyManagerInterface, "PaintBrushManipulator - could not get HotKeyManagerInterface on RegisterActions.");
+
+        // Increase Paintbrush Size
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.paintBrushManipulator.increaseSize";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = PaintbrushIncreaseSizeTitle;
+            actionProperties.m_description = PaintbrushIncreaseSizeDesc;
+            actionProperties.m_category = "PaintBrush Manipulator";
+
+            actionManagerInterface->RegisterAction(
+                EditorMainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    AdjustSize(ed_paintBrushSizeAdjustAmount);
+                }
+            );
+
+            hotKeyManagerInterface->SetActionHotKey(actionIdentifier, "]");
+        }
```

Comment:
Would it be possible to create a helper function that takes in the array of ActionOverride objects and turns them into whatever this needs? I'm concerned about all the duplicated boilerplate that this is introducing. It feels like it will be extremely easy to get the two sets of actions out of sync with each other.

Reply:
The legacy ones will be removed as soon as we enable the new system by default and start red-coding it. So the intention is to make this transition as quick as possible.

Reply:
Ohhh. I didn't realize one was replacing the other, I thought these existed in parallel for some reason. Thanks for the explanation!

Reply:
To answer the actual question... not at runtime. The legacy functions are members and require a component mode to be instantiated to be run, while the new ones are static and triggered at startup.